src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx

```diff
@@ -65,15 +66,23 @@ function BottomTabBar({selectedTab}: BottomTabBarProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {activeWorkspaceID} = useActiveWorkspace();
+    const {currentReportID} = useCurrentReportID() ?? {currentReportID: null};
     const [user] = useOnyx(ONYXKEYS.USER);
+    const [betas] = useOnyx(ONYXKEYS.BETAS);
+    const [priorityMode] = useOnyx(ONYXKEYS.NVP_PRIORITY_MODE);
     const [reports] = useOnyx(ONYXKEYS.COLLECTION.REPORT);
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
     const [reportActions] = useOnyx(ONYXKEYS.COLLECTION.REPORT_ACTIONS);
     const [transactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS);
-    const [chatTabBrickRoad, setChatTabBrickRoad] = useState<BrickRoad>(getChatTabBrickRoad(activeWorkspaceID));
+    const [chatTabBrickRoad, setChatTabBrickRoad] = useState<BrickRoad>(
+        getChatTabBrickRoad(activeWorkspaceID, currentReportID, reports, betas, policies, priorityMode, transactionViolations),
+    );
 
     useEffect(() => {
-        setChatTabBrickRoad(getChatTabBrickRoad(activeWorkspaceID));
-    }, [activeWorkspaceID, transactionViolations, reports, reportActions]);
+        setChatTabBrickRoad(getChatTabBrickRoad(activeWorkspaceID, currentReportID, reports, betas, policies, priorityMode, transactionViolations));
+        // We need to get a new brick road state when report actions are updated, otherwise we'll be showing an outdated brick road.
+        // That's why reportActions is added as a dependency here
```

Comment:
Hmm, why is this comment just about reportActions. All of these are needed since any of them can affect the RBR, right?

Reply:
@puneetlath I left the comment specific for reportActions because they are not being used inside the effect explicitly. So to prevent someone from removing the prop from the dependencies list I added that comment 😄